addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/internal/discovery/FritzBoxBridgeDiscoveryParticipant.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fritzaha.internal.discovery;
+
+import static org.openhab.binding.fritzaha.FritzAhaBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.DeviceDetails;
+import org.jupnp.model.meta.ModelDetails;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.binding.fritzaha.FritzAhaBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FritzBoxBridgeDiscoveryParticipant} is responsible for discovering
+ * new and removed FRITZ!Box devices. It uses the central
+ * {@link UpnpDiscoveryService}.
+ *
+ * @author Robert Bausdorf - Initial contribution
+ */
+public class FritzBoxBridgeDiscoveryParticipant implements
+		UpnpDiscoveryParticipant {
+	/**
+	 * Logger
+	 */
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+	/**
+	 * Provide supported thing type uid's
+	 */
+	@Override
+	public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+		return FritzAhaBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS;
+	}
+
+	/**
+	 * Create a discovery result from UPNP discovery
+	 */
+	@Override
+	public DiscoveryResult createResult(RemoteDevice device) {
+		ThingUID uid = getThingUID(device);
+		if (uid != null) {
+			logger.debug("discovered: " + device.getDisplayString());
+			Map<String, Object> properties = new HashMap<>(2);
+			properties.put(IP_ADDRESS, device.getDetails().getPresentationURI()
```

Comment:
This always returns "fritz.box", even if I am using my 546E standalone (besides a Fritz!Box), so it is effectively pointing to a wrong host :-(
What gives me the correct IP address is ```device.getIdentity().getDescriptorURL().getHost()```